### DISCUSSION TITLE
Are individual and population IDs consistent across treeseq files?

### DISCUSSION THREAD

**Discussion by clawsoon** [tskit-dev/tskit#1778] [Q&A]
Do the treeseq files at [https://zenodo.org/record/5495535](https://zenodo.org/record/5495535) all have consistent individual and population IDs?  Or is it safer to use metadata names + sources to get consistent identities?

**Comment by jeromekelleher** [comment]
Good question! @awohns I guess we should document the metadata and samples that are included somewhere and whether these IDs are stable across the files?

**Reply by awohns** [reply]
Thanks for the question @clawsoon. @jeromekelleher, metadata for individuals and populations is documented in the description at https://zenodo.org/record/5495535. These IDs are stable across the files.

**Reply by clawsoon** [reply]
Hi @awohns.  Are you saying that only the metadata IDs are stable across the files?  Or are the IDs within the treeseq files also consistent (e.g. those used by ts.individual(i) and ts.population(i) or returned by variant.genotypes())?  Thanks.

**Reply by clawsoon** [reply]
FWIW, this is what I've been using to get unique identities for individuals and populations.  Not sure if it's the best way.

```
    for individual in ts.individuals():
        individual_metadata = json.loads(individual.metadata)
        if 'sgdp_id' in individual_metadata:
            source = 'SGDP'
            individual_name = individual_metadata['sgdp_id']
        elif 'sample' in individual_metadata:
            source = 'HGDP'
            individual_name = individual_metadata['sample']
        elif 'individual_id' in individual_metadata:
            source = 'TGP'
            individual_name = individual_metadata['individual_id']

        population = ts.population(ts.node(individual.nodes[0]).population)
        population_metadata = json.loads(population.metadata)
        population_name = population_metadata['name']
        unique_population = (source, population_name)
```
